### ISSUE TITLE
Add more properties to ofVideoDevice

### ISSUE THREAD

**Issue by HalfdanJ** [openframeworks/openFrameworks#4518]
It would be great with some more fields for metadata about video devices, specifically mobile devices. 

The fields i can think of right now are 
- ```ofOrientation orientation``` the orientation of the camera relative to normal up
- ```int facing``` is the camera facing back or front

**Comment by DomAmato** [comment]
Those properties really are only features of mobile deployment though aren't they? Maybe they would be better as members of those specific grabbers since desktop applications wouldn't have such properties, unless there is something I'm missing.

**Comment by HalfdanJ** [comment]
Yes, the problem is only that the syntax for accessing the native video grabber is not very nice, so at the moment its not recommended to add special functions to specific platforms.

**Comment by arturoc** [comment]
we can probably come up with something to make the syntax for accessing the underlying grabber easiler. perhaps:

```
grabber.getGrabber<ofxAndroidVideoGrabber>().getOrientation()
```

or even

```
grabber.getMobileCamera().getOrientation()
```

where getMobileCamera could return a base class for android, ios.... something like ofBaseMobileCamera that extends ofBaseVideoGrabber and adds this methods that only make sense on a mobile device

**Comment by danthemellowman** [comment]
+1

**Comment by bakercp** [comment]
To make it consistent with video player, I'd prefer 

```
grabber.getGrabber<ofxAndroidVideoGrabber>().getOrientation()
```

Or perhaps easier, make a mobile camera class that is a subclass of ofVideoGrabber

ofMobileVideoGrabber ...

and then use 

```
grabber.getGrabber<ofMobileVideoGrabber>().getOrientation()
```

**Comment by HalfdanJ** [comment]
I like the easy accessor to the native grabber. It opens up for more interesting features. For example does the android camera (on some devices) include a face tracker, and its a better way to access advances camera settings that vary a lot between platforms.
